docs/design_docs/20230405-default_value.md

```diff
@@ -0,0 +1,80 @@
+# MEP: Default Value
+
+Current state: Under Discussion
+
+ISSUE: [[Feature]: Support Default Value #23337](https://github.com/milvus-io/milvus/issues/23337)
+
+Keywords: Default, Insert, Upsert 
+
+Released: v2.4.0
+
+## Summary
+
+Support Default Value when input data.
+
+## Motivation
+
+For now, Milvus don't support Default function. If the user pass in the same data under a certain field schema, the data can only be passed in repeatedly, which is not so flexible and user-friendly。
+
+We need a way to support Default function, which is more efficient.
+
+## Public Interfaces
+
+Add new field `default_value` in `FieldSchema`
+```proto
+message FieldSchema {
+  ...
+  repeated schema.FieldData default_value = 10;
+}
+```
+
+## Design Details
+
+1. Add the default_value in the field schema as an optional field.
+
+```proto
+message FieldSchema {
+  ...
+  schema.FieldData default_value = 10;
+}
+```
+
+2. Will use the default_value if no fields_data pass(the length of fields_data is 0 when insert and upsert).
+
+```proto
+message InsertRequest {
+  ... 
+  repeated schema.FieldData fields_data = 5;
+}
+
+message UpsertRequest {
+  ... 
+  repeated schema.FieldData fields_data = 5;
+}
+```
+
+## Compatibility, Deprecation, and Migration Plan
+
+None
+
+## Test Plan
+
+### Unit Tests
+
+- Test for using default value in proxy
+
+
+### E2E Tests
+|                  Test Cases                  |         Expected Behavior          |
+| :------------------------------------------: | :--------------------------------: |
+|         set illegal default value            |            report error            |
+|          set legal default value             |  use default value as fields data  |
+|          don't set default value             |              the same              |
```

Comment:
Add some compatibility tests, the schema built in 2.2 should be able to be used normally in the new version